applications/sudokumb/KmsDataProtectionProvider/KmsDataProtectionProvider.cs

```diff
@@ -0,0 +1,236 @@
+﻿/*
+ * Copyright (c) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.CloudKMS.v1;
+using Google.Apis.CloudKMS.v1.Data;
+using Google.Apis.Services;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Text;
+
+namespace SocialAuth.Services
+{
+    public class KmsDataProtectionProviderOptions
+    {
+        /// <summary>
+        /// Your Google project id.
+        /// </summary>
+        public string ProjectId { get; set; }
+        /// <summary>
+        /// global, us-east1, etc.

```

Comment:
worth pointing to the link where all locations are defined

Reply:
Done.